ikabot/helpers/planRoutes.py

```diff
@@ -34,17 +34,57 @@ def sendGoods(session, originCityId, destinationCityId, islandId, ships, send):
     # this can fail if a random request is made in between this two posts
     while True:
         html = session.get()
-        city = getCity(html)
-        currId = city['id']
-        data = {'action': 'header', 'function': 'changeCurrentCity', 'actionRequest': actionRequest, 'oldView': 'city', 'cityId': originCityId, 'backgroundView': 'city', 'currentCityId': currId, 'ajax': '1'}
+        current_city = getCity(html)  # the city the bot is right now
+        city = getCity(session.get(city_url + originCityId))  # the origin city
+        currId = current_city['id']
+
+        # Change from the city the bot is sitting right now to the city we want to load resources from
+        data = {
+            'action': 'header',
+            'function': 'changeCurrentCity',
+            'actionRequest': actionRequest,
+            'oldView': 'city',
+            'cityId': originCityId,
+            'backgroundView': 'city',
+            'currentCityId': currId,
+            'ajax': '1'
+        }
+
         session.post(payloadPost=data)
 
-        data = {'action': 'transportOperations', 'function': 'loadTransportersWithFreight', 'destinationCityId': destinationCityId, 'islandId': islandId, 'oldView': '', 'position': '', 'avatar2Name': '', 'city2Name': '', 'type': '', 'activeTab': '', 'transportDisplayPrice': '0', 'premiumTransporter': '0', 'minusPlusValue': '500', 'capacity': '5', 'max_capacity': '5', 'jetPropulsion': '0', 'transporters': ships, 'backgroundView': 'city', 'currentCityId': originCityId, 'templateView': 'transport', 'currentTab': 'tabSendTransporter', 'actionRequest': actionRequest, 'ajax': '1'}
+        # data = {'action': 'transportOperations', 'function': 'loadTransportersWithFreight', 'destinationCityId': destinationCityId, 'islandId': islandId, 'oldView': '', 'position': '', 'avatar2Name': '', 'city2Name': '', 'type': '', 'activeTab': '', 'transportDisplayPrice': '0', 'premiumTransporter': '0', 'minusPlusValue': '500', 'capacity': '5', 'max_capacity': '5', 'jetPropulsion': '0', 'transporters': ships, 'backgroundView': 'city', 'currentCityId': originCityId, 'templateView': 'transport', 'currentTab': 'tabSendTransporter', 'actionRequest': actionRequest, 'ajax': '1'}
```

Comment:
If this code is no longer used, can it be deleted?

Reply:
Done!